data/data/azure/main.tf

```diff
@@ -36,6 +36,7 @@ module "bootstrap" {
   tags                = local.tags
   storage_account     = azurerm_storage_account.cluster
   nsg_name            = module.vnet.master_nsg_name
+  private             = module.vnet.private
```

Comment:
This goes along with my other comment about passing a `private` bool instead of a `publish` string. I think it would be nicer here because we could just pass `var.private`. `module.vnet.private` here creates too much of a dependency between bootstrap and vnet for just processing a variable.